airbyte-integrations/connectors/source-file/source_file/client.py

```diff
@@ -350,7 +350,7 @@ def dtype_to_json_type(current_type: str, dtype) -> str:
             return current_type
         if dtype == object:
             return "string"
-        if dtype in number_types and (not current_type or current_type in number_types):
+        if dtype in number_types and (not current_type or current_type == "number"):
```

Comment:
@Alihassanc5 I think the changes you're submitting are not improving the connector.
The variable `number_types = ("int64", "float64")` already transform int and floats. Those are dtypes from pandas.

Reply:
@marcosmarxm Sorry for the confusion, let's iterate this code for 2 chunks of CSV file.

During the first chunk, the current_type is None, and the dtype is "int64". As expected, the code correctly identifies this as a "number" type.

However, during the second chunk, the current_type is "number", but the dtype is still "int64". At this point, the code incorrectly compares the current_type with "number_types" instead of previous column type "number". This leads to the code setting the column type to "string" instead of "integer" for over 10,000 records.

To fix this issue, the code needs to be modified to compare the current_type with "number" instead of "number_types". This will ensure that the column type is correctly identified as an integer for all integer values in the CSV file.

Reply:
@Alihassanc5 Can you add a unit test to validate the change?

Reply:
@marcosmarxm I have updated the unit test to validate this change.
current_type, d_type, expected

("number", "int64", "number"),
("number", "float64", "number"),

Reply:
@marcosmarxm any update regarding this PR?